packages/frontend/src/state/lp/hooks.ts

```diff
@@ -177,6 +178,159 @@ export const useCollectFees = () => {
   return collectFees
 }
 
+// Rebalance via general swap
+export const useRebalanceGeneralSwap = () => {
+  const address = useAtomValue(addressAtom)
+  const controllerHelperContract = useAtomValue(controllerHelperHelperContractAtom)
+  const { controllerHelper, weth, oSqueeth, squeethPool } = useAtomValue(addressesAtom)
+  const controllerContract = useAtomValue(controllerContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const isWethToken0 = useAtomValue(isWethToken0Atom)
+  const getDebtAmount = useGetDebtAmount()
+  const getVault = useGetVault()
+  const getDecreaseLiquidity = useGetDecreaseLiquidity()
+  const getPosition = useGetPosition()
+  const getTwapSqueethPrice = useGetTwapSqueethPrice()
+  const squeethPoolContract = useAtomValue(squeethPoolContractAtom)
+  const getQuote = useGetQuote()
+  const rebalanceGeneralSwap = useAppCallback(
+    async (vaultId: number, lowerTickInput: number, upperTickInput: number, onTxConfirmed?: () => void) => {
+      const vaultBefore = await getVault(vaultId)
+      const uniTokenId = vaultBefore?.NFTCollateralId 
+      const position = uniTokenId ? await getPosition(uniTokenId) : null
+      if (!controllerContract || !controllerHelperContract || !address || !position || !vaultBefore || !squeethPoolContract) return
+      const shortAmount = fromTokenAmount(vaultBefore.shortAmount, OSQUEETH_DECIMALS)
+      const debtInEth = await getDebtAmount(shortAmount)
+      const collateralToFlashloan = debtInEth.multipliedBy(COLLAT_RATIO)
+
+      const amount0Min = new BigNumber(0)
+      const amount1Min = new BigNumber(0)
+
```

Comment:
We might want to change these to something close to expected values to avoid getting sandwiched

Reply:
Yep, planning to use:

`amount0Min = amount0 * (1 - slippage)`
`amount1Min = amount1 * (1 - slippage)`

Does that sound good?

Reply:
sounds excellent - slippage as input param?

Reply:
Yep